### PR TITLE
Reduce dart analysis warnings and hints and remove "Interface" classes (e.g. `RealtimeInterface`)

### PR DESCRIPTION
**Pull request by ben-xD**
This PR doesn't add any new features.

- It fixes https://github.com/ably/ably-flutter/issues/169. Importing files were previously annoying
- it deletes unnecessary interface classes: e.g. `RealtimeInterface`: It was confusing to see some interface classes being used, others being completely neglected and its concrete implementation used everywhere instead. 
  - I removed the notes in the `CONTRIBUTING.md` in 9e4a5a7e1b13ad3332cf09a282b68d2685a584aa:
   > The `Realtime` 'interface' is an [abstract class](https://dart.dev/guides/language/language-tour#abstract-classes) which seems to be the correct way to do this in Dart - every class has an ['implicit interface'](https://dart.dev/guides/language/language-tour#implicit-interfaces) and then you make the class abstract to present it as a more traditional interface as understood by the perhaps a Java programmer.
  - Explanation: All of the `XInterface` are unnecessary. Dart classes have an implicit interface, so this means anyone who wants to implement the interface can just implement the interface of the implementation. For example, we have a `Animal` class with some methods and abstract methods. `Cat` can either extend or implement `Animal`, depending on if it wants to inherit from Animal (and get its method implementations), or just conform to its interface where it will have to implement the methods itself. 
  - More importantly, `Realtime` is a concrete class, and can be used directly, without needing a `RealtimeInterface`, `ConnectionInterface`, `PaginatedResultInterface`, `RealtimeChannelInterface`, `RealtimeChannelsInterface`, `RealtimePresenceInterface`, `RestChannelInterface`, `RestChannelsInterface`, `RestInterface` and `RestPresenceInterface`. There is no need for a `RealtimeInterface`.
  - In the context of Federated Plugins, you might think that these interfaces classes (abstract classes) could be moved to the "platform_interface" package, however that is quite far down the line and quite open to discussion. We are solving a problem in the codebase today: complexity/confusion.
- Includes other small fixes to reduce dart analysis warnings and hints. (e.g. using `const` keyword)

Basically, this PR handles technical debt and pain points I have faced over a few months of development in the library.

### CODE REVIEW

**Review by ikbalkaya — APPROVED** [review]
Looks good to me

File: lib/src/crypto/src/crypto.dart [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
-  static const KEY_LENGTH_128_BITS = 128;
-  static const DEFAULT_MODE = 'cbc';
-  static const KEY_TYPE_ERROR_MESSAGE =
+  static const defaultAlgorithm = 'aes';
+  static const defaultBlockLengthInBytes = 16;
+  static const defaultKeyLengthInBits = 256;
+  static const keyLength128bits = 128;
+  static const defaultMode = 'cbc';
+  static const keyTypeErrorMessage =
       'A key must either be a String or a Uint8List.';
```
[the remark this answers is not in the document]
Comment (ben-xD) [reply lib/src/crypto/src/crypto.dart]: Cool, I've done that in d0bd890bff8dab33703b22a22d5a8189b85ee19b

**Review by QuintinWillison — COMMENTED** [review]
Given my current workload and the size of this pull request, I'm not going to be able to authentically review it with any amount of value.

I recognise that from your perspective this is something of a 'roll up' set of fixes for concerns you've encountered, but from my point of view there is little to focus on with lots of concerns conflated. I need to simply trust your judgement that the changes presented here move the codebase forward in a positive way and that they don't break anything.
